FLEDGE_extended_PA_reporting.md

```diff
@@ -255,6 +256,82 @@ The `reserved.always` event-type is a special value that indicates a report shou
 matter whether a bid wins the auction or not. The browser will always trigger reporting for the
 `always` contributions.
 
+## Per-participant metrics
+
+More recent versions of Chrome offer some additional features, focused on measuring resource
+usage and whole-auction metrics rather than those specific to a particular function execution
+or the winning bid.
+
+First, the new `reserved.once` event-type is a special value that, for each sub-auction, selects a
+random invocation of `generateBid()` and of `scoreAd()` independently, and reports private
+aggregation contributions with that event only from those executions. (In case of a multi-seller
+auction, the top-level auction will have a single `scoreAd()` invocation selected as well).
+
+The event may not be used in `reportWin()` or `reportResult()`; since those already run at most
+once per auction level, `reserved.always` may be used instead.
+
+This feature is intended for reporting overall per-participant metrics only once rather than for
+every interest group participating. A number of new `baseValues` representing such values are
+available and described below, but it can also be useful with per-IG metrics which are not expected to
+vary much like `signals-fetch-time`, to sample them.
+
+While usage of `reserved.once` will be ignored by older versions, newly added `baseValues` will not
+be, so the calls to `contributeToHistogramOnEvent()` should be individually wrapped in `try/catch`.
+That is also encouraged in general since `contributeToHistogramOnEvent()` is specified to throw
+on permission policy violations.
+
+Users using this are strongly encouraged to report their metrics during the beginning of their
+scripts, since if the script hits a per script time out before asking to report them nothing will
+get sent, which can result in inaccuracy, especially for `percent-scripts-timeout`.
+
+### Per-participant base values.
+
+The newly added base values are as following:
+* `participating-ig-count`: number of interest groups that got a chance to participate in this
+  (sub)auction, i.e. they had registered ads, and if no timeouts were to hit, their `generateBid()`
+  would have been run.
+* `average-code-fetch-time`: average time of all code fetches (including JavaScript and WASM) used
+  for all invocations of particular function type for given (sub)auction.
+* `percent-scripts-timeout`: percentage of script executions of this kind that hit the per-script
+  timeout.
```

Comment:
>percentage of script executions of this kind that hit the per-script timeout

This _of this kind_ reading would be accessible in reporting functions via the suggested use of `reserved.always`.

Reply:
Not sure what you mean here?

Reply:
Poor wording as I wrapped my head around values not strictly associated with `reserved.once`.

>`percent-scripts-timeout`: percentage of script executions of this kind that hit the per-script timeout.

My comment was recognizing that one must use `reserved.always` in a reporting function. Since this baseValue actually measures invocations of the function in which the histogram contribution is reported, might the wording be "of this function?"

>while percent-scripts-timeout in `reportWin()` is either 0 or 100 dependent on whether the reporting function's execution timed out or not (assuming reporting is done early enough to happen if it did)

This applies to `scoreAd()` also, right? Maybe "in reporting functions?"

<br/>

>`average-code-fetch-time`: average time of all code fetches (including JavaScript and WASM) used for all invocations of particular function type for given (sub)auction.

I read this as _different measurement_ by function type (bid/score versus reporting), but until Chrome introduces a separate reporting logic url ([#679](https://github.com/WICG/turtledove/issues/679#issuecomment-1703973736)) there's no differentiation, right? Because the fetches are part of the executor setup. 

To verify, whether using `.once` or `.always` (in reportResult), sellers will receive the same value - their scoringLogicURL's code fetch time.

Buyers would also receive the same value regardless of where the contribution is reported.  
A buyer with IGs using only biddingLogicURLs would receive the average code fetch across IGs that made it that far. 
A buyer with IGs using WASM as well would receive the average code fetch irrespective of type.

Reply:
So reporting code fetch time being the same as for generateBid/scoreAd is not not necessarily true, since we have a process limit, and so the process may actually need to be started fresh for reporting (and fetch fresh... hopefully from cache). That's actually the common case for reportWin (basically unless a concurrent auction keeps the process alive).

Reply:
Oh, yes, right. You don't hold onto that [limited resource](https://github.com/WICG/turtledove/blob/main/PA_implementation_overview.md#:~:text=There%20are%20separate,to%20resource%20constraints.
). 

>There are separate Chrome-wide limits on the number of buyer processes (10) and seller processes (3).  These numbers are not optimized, and subject to change.  On mobile, Chrome will generally reuse the single renderer process with one thread for bidders and one for sellers, instead of using more secure, isolated processes, due to resource constraints.

Verifying that "on mobile" equates to 
```
Sec-CH-UA-Mobile: ?1
```

Reply:
No, it equates to "Android" (so telling devtools to emulate an iPhone on your laptop won't change that... not would "request desktop site" on a phone). I suppose it should probably just say that.